### PR TITLE
FOGL-7678_patch: Show service dropdown list group by and labeled by type

### PR DESCRIPTION
**Pull request by monika-Dianomic**

### CODE REVIEW

File: src/app/components/core/control-dispatcher/pipelines/add-pipeline-filter/add-pipeline-filter.component.html [hunk 1]
```diff
@@ -2,8 +2,8 @@
   <div class="modal-background"></div>
   <div class="modal-card">
       <header class="modal-card-head">
-      <span *ngIf="pipelineName != undefined" class="modal-card-title is-size-6">{{pipelineName}} - Add Filter
```
Comment (praveen-garg) [inline src/app/components/core/control-dispatcher/pipelines/add-pipeline-filter/add-pipeline-filter.component.html:5]: I think is-size-6 should be here, same as other card title.

File: src/app/components/core/control-dispatcher/pipelines/add-pipeline-filter/add-pipeline-filter.component.html [hunk 2]
```diff
@@ -2,8 +2,8 @@
...
-        <p class="help is-6 has-text-grey">Control Pipeline</p>
+      <span *ngIf="pipelineName != undefined" class="modal-card-title">{{pipelineName}} - Add Filter
+        <p class="help has-text-grey ctrl-pipeline">Control Pipeline</p>
```
[the remark this answers is not in the document]
Comment (monika-Dianomic) [reply src/app/components/core/control-dispatcher/pipelines/add-pipeline-filter/add-pipeline-filter.component.html:6]: I tried `pt-2` but it was not working. That’s why I had to use custom css here.

File: src/app/components/core/control-dispatcher/pipelines/add-pipeline/add-control-pipeline.component.css [hunk 3]
```diff
@@ -114,4 +114,36 @@
   [hidden] {
     display: none !important;
   }
+
+  .filter-row {
+    margin-bottom: 0;
```
[the remark this answers is not in the document]
Comment (monika-Dianomic) [reply src/app/components/core/control-dispatcher/pipelines/add-pipeline/add-control-pipeline.component.css:119]: I tried `mb-0` also but it was also not working.

File: src/app/components/core/control-dispatcher/pipelines/add-pipeline/add-control-pipeline.component.ts [hunk 4]
```diff
@@ -164,6 +165,14 @@ export class AddControlPipelineComponent implements OnInit {
     this.cdRef.detectChanges();
   }
 
+  ngAfterViewInit() {
+    setTimeout(() => {
```
[the remark this answers is not in the document]
Comment (monika-Dianomic) [reply src/app/components/core/control-dispatcher/pipelines/add-pipeline/add-control-pipeline.component.ts:169]: It is being used to wait for the form to load completely. If I don't use timeout, it doesn't get any value in `this.pipelineForm.form.controls['name']`. `undefined` error occurred in that case.

File: src/app/components/core/control-dispatcher/pipelines/add-pipeline/add-control-pipeline.component.ts [hunk 5]
```diff
@@ -552,8 +595,10 @@ export class AddControlPipelineComponent implements OnInit {
           /** request completed */
           this.ngProgress.done();
           data.forEach(asset => {
-            this.destinationNameList.push(asset.assetCode);
+            asset['name'] = asset.assetCode;
```
[the remark this answers is not in the document]
Comment (monika-Dianomic) [reply src/app/components/core/control-dispatcher/pipelines/add-pipeline/add-control-pipeline.component.ts:598]: Yes, I have assigned assetCode value to `name` property as I am showing value of `name` property in the dropdown. And it is common for all.

File: src/app/components/core/control-dispatcher/pipelines/add-pipeline/add-control-pipeline.component.css [hunk 6]
```diff
@@ -114,4 +114,36 @@
...
+  }
+
+  .accordion.card {
```
[the remark this answers is not in the document]
Comment (monika-Dianomic) [reply src/app/components/core/control-dispatcher/pipelines/add-pipeline/add-control-pipeline.component.css:122]: not working

**Review by hemantgarg923 — APPROVED** [review]

**Pull request merged**